eng/common/pipelines/templates/steps/check-duplicate-file-name.yml

```diff
@@ -0,0 +1,25 @@
+parameters:
+  ServiceDirectory: ''
+steps:
+- pwsh: |
+    $scannedDirectory = @{}
+    if ("$(Build.Reason)" -eq 'PullRequest') {
+      $changedFiles = & "eng/common/scripts/get-changedfiles.ps1"
```

Comment:
We should avoid trying to traverse the files on disk because that will give different answers depending on OS settings. I suggest looking at https://git-scm.com/docs/git-ls-files which should list all the files in the git database and if you find more duplicates while ignoring case then fail. Give something like `git ls-files | Group-Object | Where-Object { $_.Count -gt 1 } | Select-Object Name` a try.